### ISSUE TITLE
Broken seasonal gear can show up as Gem-purchasable in Featured Items

### ISSUE THREAD

**Issue by SabreCat** [HabitRPG/habitica#10719]
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found from the Habitia website at User Icon > Settings > API)
### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)

As reported in the bug guild:

>**Il Mago** (c073342f-4a65-4a13-9ffd-9e7fa5410d6b) tier 7 Legendary Socialite
>for the series of things that you have already bought for gems but you can buy again for gems (equipment)
>
>current situation: I have a lot of equipment that I previously owned but was then removed from my account so I could buy it back for gold
>
>bug: I have in the seasonal shop as Featured Set Mummy Medic (Healer), with a gem price, but I already have bought this for gems so I should be able to buy it for gold (in the Market it appears correctly)
>
>this could be an other thing to fix before the "empty the armoire" feature go live as it will be more common to have as featured set something that you have already acquired in some way and then removed by the feature

Basically, if a user has a seasonal equipment item in `false` status (i.e., "broken" due to Health knockout, or removed due to legacy Orb of Rebirth behavior), and that item is featured in the Seasonal Shop, it'll show up to be bought with Gems. It should instead show as Gold-buy.

**Comment by kirsty-tortoise** [comment]
Given that non-featured items are not currently visible in the seasonal shop at all if they were previously owned, would this be reasonable for featured items as well?

Or should both featured and non-featured items in the seasonal shop be purchasable for gold if they've been purchased and then lost?

**Comment by SabreCat** [comment]
@kirsty-tortoise "should both featured and non-featured items in the seasonal shop be purchasable for gold if they've been purchased and then lost?" -- I'd say yes!

**Comment by citrusella** [comment]
Since this appears to be an identical issue to #13923 which was closed as fixed by #14326 should it also be closed?

**Comment by CuriousMagpie** [comment]
Thanks for the catch!, @citrusella!

We can always re-open this issue if it hasn't been fixed by #14326.
